Modules/_elementtree.c

```diff
@@ -1213,12 +1213,8 @@ _elementtree_Element_extend_impl(ElementObject *self, PyTypeObject *cls,
     PyObject* seq;
     Py_ssize_t i;
 
-    seq = PySequence_Fast(elements, "");
```

Comment:
Ideally we want to keep the message about the incorrect type.

Reply:
There is something I'm actually wondering. Should we actually accept any iterable or just sequences? because here the message says "expected sequence" and not "expected iterable". 

Anyway, your argument is sound and consistent with other occurrences of `PySequence_Fast` (except for the one in dictobject but this one is special I think)

Reply:
I believe any iterable, per all other usages of PySequence, the typeshed and the general commonplace usage of `.extend()` methods.

I haven't seen anywhere in the code where a sequence is required before the implicit conversion to PySequence. The pure Python implementation also accepts arbitrary iterables, which currently means there's a deviation between the Python and C code in case of errors.

IIRC the docs do state sequence but I can change the docs to be more lenient. It is very common for it to write sequences in places where any iterables are accepted so I haven't touched it.

As it is more permissive, no bugs are relevant except for potential inheritance specifically from the C implementation breaking LSP, but the interface according to the shed is any Iterable anyway.

All in all, yes, I believe the bug is a bug :-)

Thanks for reviewing btw 🐍

Reply:
As the biggest relevant external package that is also a drop-in replacement, I've checked lxml too and it seems that they accept an iterable. It further hints towards an iterable being the expected behavior.

Let's say like this - if we change it to a sequence only, we're going to break a lot of code, and we'll require a more specific type for no apparent reason.

If we change it to an arbitrary iterable, the odds of breaking any code is astronomically low.

Reply:
Thanks for the investigation. I don't mind amending the docs since we are anyway accepting any iterable (and not just sequences). You'll need to wait for a core dev to review your PR though since I don't have commit permissions.

Reply:
Amended the docs per your suggestion.

Haven't added a versionchanged tag as it's technically a fix and clarification. We accepted iterables all along, just with a bug in case of an exception. 

Thank you for your review and suggestions.
If you believe it is ready you can accept the review so the core-reviewer tag will be added by the bot.

Reply:
I'll let this one opened so that other devs could discuss here.